src/core/trtserver2.h

```diff
@@ -0,0 +1,340 @@
+/// \param inference_request The request object.
+/// \param timeout_us The timeout, in microseconds.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error*
+TRTSERVER2_InferenceRequestSetTimeoutMicroseconds(
+    TRTSERVER2_InferenceRequest* inference_request, uint64_t timeout_us);
+
+/// Add an input to a request.
+/// \param inference_request The request object.
+/// \param name The name of the input.
+/// \param datatype The type of the input. Valid type names are BOOL,
+/// UINT8, UINT16, UINT32, UINT64, INT8, INT16, INT32, INT64, FP16,
+/// FP32, FP64, and BYTES.
+/// \param shape The shape of the input.
+/// \param dim_count The number of dimensions of 'shape'.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER2_InferenceRequestAddInput(
+    TRTSERVER2_InferenceRequest* inference_request, const char* name,
+    const char* datatype, const int64_t* shape, uint64_t dim_count);
+
+/// Remove an input from a request.
+/// \param inference_request The request object.
+/// \param name The name of the input.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER2_InferenceRequestRemoveInput(
+    TRTSERVER2_InferenceRequest* inference_request, const char* name);
+
+/// Remove all inputs from a request.
+/// \param inference_request The request object.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER2_InferenceRequestRemoveAllInputs(
+    TRTSERVER2_InferenceRequest* inference_request);
+
+/// Assign a buffer of data to an input. The buffer will be appended
+/// to any existing buffers for that input. The 'inference_request'
+/// object takes ownership of the buffer and so the caller should not
+/// modify or free the buffer until that ownership is released by
+/// 'inference_request' being deleted or by the input being removed
+/// from 'inference_request'.
+/// \param inference_request The request object.
+/// \param name The name of the input.
+/// \param base The base address of the input data.
+/// \param byte_size The size, in bytes, of the input data.
+/// \param memory_type The memory type of the input data.
+/// \param memory_type_id The memory type id of the input data.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER2_InferenceRequestAppendInputData(
+    TRTSERVER2_InferenceRequest* inference_request, const char* name,
+    const void* base, size_t byte_size, TRTSERVER_Memory_Type memory_type,
+    int64_t memory_type_id);
+
+/// Clear all input data from an input, releasing ownership of the
+/// buffer(s) that were appended to the input with
+/// TRTSERVER2_InferenceRequestAppendInputData.
+/// \param inference_request The request object.
+/// \param name The name of the input.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER2_InferenceRequestRemoveAllInputData(
```

Comment:
So to change the input data, user will call `RemoveAllInputData` first and then call the append above?

Reply:
If you want to use the same input buffers you can just change the data in place and then run inference again. If you want to set an input to use a different buffer(s) then you must remove-all and then append the new buffers.